src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -840,6 +840,24 @@ public function findDefinition($id)
      */
     public function createService(Definition $definition, $id, $tryProxy = true)
```

Comment:
So.. as this is public API should we deprecate calling it as well before compiling? ^^ Opposed to #20533 of course ;-) Including `resolveServices` that is, as right now a reference will trigger the deprecation ([ref](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/ContainerBuilder.php#L952)), whereas a definition doesnt.

Imo. we should move forward with this, but i do agree with you in terms of providing an alternative solution. Lets focus on that?